pkgs/applications/graphics/pdfcpu/default.nix

```diff
@@ -2,24 +2,56 @@
 
 buildGoModule rec {
   pname = "pdfcpu";
-  version = "0.5.0";
+  version = "0.6.0";
 
   src = fetchFromGitHub {
     owner = "pdfcpu";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-dEAlOKjNXL7zqlll6lqGmbopjdplDR3ewMMNu9TMsmw=";
+    hash = "sha256-410DIdsPXKMxnBL+1KX/DIPktqoB/ZxgF5GXEiqLcbw=";
+    # Apparently upstream requires that the compiled executable will know the
+    # commit hash and the date of the commit. This information is also presented
+    # in the output of `pdfcpu version` which we use as a sanity check in the
+    # installCheckPhase. This was discussed upstream in:
+    #
+    # - https://github.com/pdfcpu/pdfcpu/issues/751
+    # - https://github.com/pdfcpu/pdfcpu/pull/752
+    #
+    # The trick used here is to write that information into files in `src`'s
+    # `$out`, and then read them into the `ldflags`. We also delete the `.git`
+    # directories in `src`'s $out afterwards, imitating what's done if
+    # `leaveDotGit = false;`
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      git rev-parse HEAD > $out/COMMIT
+      git log -1 --pretty=%cd --date=format:'%Y-%m-%dT%H:%M:%SZ' > $out/SOURCE_DATE_EPOCH
+      find "$out" -name .git -print0 | xargs -0 rm -rf
+    '';
   };
 
-  vendorHash = "sha256-WZsm2wiKedMP0miwnzhnSrF7Qw+jqd8dnpcehlsdMCA=";
+  vendorHash = "sha256-jVX/CFf9dd9qD3gyGVDjbfINtPLCsgdjWFix4BCpDZ0=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=v${version}"
+  ];
+
+  # ldflags based on metadata from git and source
+  preBuild = ''
+    ldflags+=" -X main.commit=$(cat COMMIT)"
+    ldflags+=" -X main.date=$(cat SOURCE_DATE_EPOCH)"
+  '';
+
 
   # No tests
   doCheck = false;
   doInstallCheck = true;
   installCheckPhase = ''
     export HOME=$(mktemp -d)
     echo checking the version print of pdfcpu
-    $out/bin/pdfcpu version | grep ${version}
+    $out/bin/pdfcpu version | grep -Eo '^(pdfcpu: v${version}|commit: $(cat COMMIT) \($(cat SOURCE_DATE_EPOCH)\))'
```

Comment:
IIRC the `|` regex operand means match 'either' of the pattern to the left and to the right. If so, that's not ideal, as it means the pattern could match due to the version, while the other pattern doesn't match. It's OK to simply run the `pdfcpu version | grep` 3 times, and match each of the strings in each run.

Reply:
You are absolutely right! Thanks for pointing this out. Since `pdfcpu version` output shows the commit hash and date on one line two times should be enough.